ui-tests/cypress/integration/components/ux_fragments.spec.js

```diff
@@ -1,141 +1,783 @@
-import {generateRandomId} from '../../support/utils';
-import HomePage           from '../../support/pageObjects/HomePage';
+import {generateRandomId, generateRandomTypeCode} from '../../support/utils';
+import {htmlElements}                             from '../../support/pageObjects/WebElement';
 
 describe('UX Fragments', () => {
 
-  let currentPage;
-
   beforeEach(() => {
     cy.kcAPILogin();
     cy.kcUILogin('login/admin');
+    cy.wrap(null).as('fragmentToBeDeleted');
+    fragment.code = `A${generateRandomId()}`;
   });
 
-  afterEach(() => cy.kcUILogout());
-
-  describe([Tag.GTS], 'Base tests', () => {
-
-    beforeEach(() => {
-      fragment.code = `${generateRandomId()}`;
-      cy.wrap(null).as('fragmentToBeDeleted');
+  afterEach(() => {
+    cy.kcUILogout();
+    cy.get('@fragmentToBeDeleted').then(fragmentToBeDeleted => {
+      if (fragmentToBeDeleted !== null) {
+        cy.fragmentsController().then(controller => controller.deleteFragment(fragmentToBeDeleted.code));
+      }
     });
+  });
 
-    afterEach(() => {
-      cy.get('@fragmentToBeDeleted').then(fragmentToBeDeleted => {
-        if (fragmentToBeDeleted !== null) {
-          cy.fragmentsController().then(controller => controller.deleteFragment(fragmentToBeDeleted.code));
-        }
-      });
+  describe('Fragments pages visualization', () => {
+
+    it([Tag.SMOKE, 'ENG-3522'], 'Ux FragmentsPage is displayed', () => {
+      openFragmentsPage()
+          .then(page => {
+            page.getContent().getTable()
+                .should('exist').and('be.visible');
+            page.getContent().getTableHeaders().should('have.length', 4)
+                .then(elements => cy.validateListTexts(elements, ['Code', 'Widget type', 'Plugin', 'Actions']));
+
+            page.getContent().getPagination().get()
+                .should('exist').and('be.visible');
+            page.getContent().getPagination().getInput()
+                .should('have.value', 1);
+
+            page.getContent().getSearchForm()
+                .should('exist').and('be.visible');
+            page.getContent().getAddButton()
+                .should('exist').and('be.visible');
+          });
     });
-
-    it('Add a new fragment', () => {
-      currentPage = openFragmentsPage();
-
-      currentPage = currentPage.getContent().openAddFragmentPage();
+    it([Tag.SMOKE, 'ENG-3522'], 'AddPage is displayed', () => {
+      openFragmentsPage()
+          .then(page =>
+              page.getContent().openAddFragmentPage());
       cy.validateUrlPathname('/fragment/add');
-      currentPage.getContent().typeCode(fragment.code);
-      currentPage.getContent().typeGuiCode(fragment.guiCode);
-      currentPage = currentPage.getContent().save();
-      cy.wrap(fragment).as('fragmentToBeDeleted');
 
-      cy.validateToast(currentPage);
-    });
+      cy.get('@currentPage')
```

Comment:
why is this not concatenated with the previous operation? is `openAddFragmentPage` not updating and returning _currentPage_?

Reply:
there was a problem with the validateUrl command. I fixed it :)